src/fetch/Dockerfile

```diff
@@ -0,0 +1,29 @@
+# Use a Python image with uv pre-installed
+FROM ghcr.io/astral-sh/uv:python3.12-bookworm-slim
+
+# Install the project into `/app`
+WORKDIR /app
+
+# Enable bytecode compilation
+ENV UV_COMPILE_BYTECODE=1
+
+# Copy from the cache instead of linking since it's a mounted volume
+ENV UV_LINK_MODE=copy
+
+# Install the project's dependencies using the lockfile and settings
+RUN --mount=type=cache,target=/root/.cache/uv \
+    --mount=type=bind,source=uv.lock,target=uv.lock \
+    --mount=type=bind,source=pyproject.toml,target=pyproject.toml \
+    uv sync --frozen --no-install-project --no-dev
```

Comment:
Any particular reason we use --no-install-project here? The alternative would be to do that and run just call `mcp-server-fetch` since our PATH already points to the .venv. This would avoid having to go through uvx. I am fine with this, but i am curious if there is a reason for one over the other.

Reply:
@dsp-ant what you're saying here makes sense. We could potentially create a uv-less python environment and just copy the virtual environment from this stage into a small uv-less final stage.

Reply:
@dsp-ant so my summary here is that I think we should keep the uv stage running with two layers (one slow moving one that does not install the project) and a second faster moving one with the project.  This will be a lot faster when we pick up changes.

Your suggestion to run without uv is great.  In the latest Dockerfile, the final stage is uv-less.